app/javascript/controllers/color_picker_controller.js

```diff
@@ -0,0 +1,218 @@
+    
+    this.iconTargets.forEach(icon => {
+      const iconWrapper = icon.nextElementSibling;
+      iconWrapper.style.removeProperty("background-color")
+      iconWrapper.style.color = "black";
+      iconWrapper.classList.add(`hover:bg-[${currentColor}]`)
+    });
+
+    this.updateSelectedIconColor(currentColor);
+  }
+
+  handleIconChange(e) {
+    const iconSVG = e.currentTarget.closest('label').querySelector('svg').cloneNode(true);
+    this.avatarTarget.innerHTML = '';
+    iconSVG.style.padding = "0px"
+    this.avatarTarget.appendChild(iconSVG);
+  }
+
+  updateSelectedIconColor(color) {
+    if (this.selectedIcon) {
+      const iconWrapper = this.selectedIcon.nextElementSibling;
+      iconWrapper.style.backgroundColor = `color-mix(in srgb, ${color} 15%, white)`;
+      iconWrapper.style.color = color;
+    }
+  }
+
+  handleColorChange(e) {
+    const color = e.currentTarget.value;
+    this.colorInputTarget.value = color;
+    this.colorPreviewTarget.style.backgroundColor = color;
+    this.updateAvatarColors(color);
+    this.updateSelectedIconColor(color);
+  }
+
+  handleContrastValidation(contrastRatio, rgbacolor) {
+    if (contrastRatio < 4.5) {
+      this.colorInputTarget.setCustomValidity("Poor contrast, choose darker color or auto-adjust.");
+
+      if (this.paletteSectionTarget.querySelector("span")) return;
+
+      const darkColor = this.darkenColor(rgbacolor).toString();
+      
+      const span = document.createElement("span");
+      span.style.color = "var(--color-destructive)";
+      span.style.alignSelf = "start";
+      span.classList.add("text-sm");
+      span.innerHTML = "Poor contrast, choose darker color or ";
+
+      const button = document.createElement("button");
+      button.textContent = "auto-adjust.";
+      button.type = "button";
+      button.style.textDecoration = "underline";
+      button.style.cursor = "pointer";
+
+      button.addEventListener("click", () => {
+        this.colorInputTarget.value = darkColor;
+        this.colorInputTarget.dataset.colorPickerColorValue = darkColor;
+        this.colorPreviewTarget.style.backgroundColor = darkColor;
+        this.picker.setColor(darkColor);
+        this.colorInputTarget.setCustomValidity("");
+
+        const span = this.paletteSectionTarget.querySelector("span");
+        if (span) span.remove();
+      });
+
+      span.appendChild(button);
+      this.paletteSectionTarget.append(span);
+    } else {
+      this.colorInputTarget.setCustomValidity("");
+      const span = this.paletteSectionTarget.querySelector("span");
+      if (span) span.remove();
+    }
+  }
+
+  backgroundColor([r,g,b,a], percentage) {
+    const mixedR = Math.round((r * (percentage / 100)) + (255 * (1 - percentage / 100)));
+    const mixedG = Math.round((g * (percentage / 100)) + (255 * (1 - percentage / 100)));
+    const mixedB = Math.round((b * (percentage / 100)) + (255 * (1 - percentage / 100)));
+    return [mixedR, mixedG, mixedB];
+  }
```

Comment:
We have several hardcoded values of `5%`, `10%`, and `15%` throughout the controller.  We should add a private method:

```js
#backgroundColor(color) {
  return `color-mix(in oklab, ${color} 10%, transparent)`;
}
```

And then throughout the controller:

- Remove all borders (those are outdated from prev designs)
- Refer to this private method for avatar + icon backgrounds